src/session_impl.cpp

```diff
@@ -6894,8 +6901,7 @@ namespace {
 		// there's no point in starting the UPnP mapper for a network that isn't
 		// connected to the internet. The whole point is to forward ports through
 		// the gateway
-		if ((s->flags & listen_socket_t::local_network)
-			|| (s->flags & listen_socket_t::proxy))
+		if ((s->flags & listen_socket_t::local_network))
```

Comment:
here too. You're removing this check. I would imagine you want to take the new option into account instead